File: 01689. Partitioning Into Minimum Number Of Deci-Binary Numbers/01689.go
// https://leetcode.com/problems/partitioning-into-minimum-number-of-deci-binary-numbers

func minPartitions(n string) int {
    var maxDigit rune

    for _, digit := range n {
        if maxDigit < digit {
            maxDigit = digit

            if maxDigit == '9' {
                break
            }
        }
    }

    return int(maxDigit - '0')
}

File: LeetCode/01689. Partitioning Into Minimum Number Of Deci-Binary Numbers/01689.c
int minPartitions(char * n) {
    char c, maxDigit = '\0';
    
    while ((c = *n) != '\0') {
        maxDigit = (maxDigit > c) ? maxDigit : c;
        ++n;
    }

    return maxDigit - '0';
}
